x/posts/internal/keeper/keeper.go

```diff
@@ -53,7 +53,9 @@ func (k Keeper) SavePost(ctx sdk.Context, post types.Post) {
 	store.Set([]byte(types.PostStorePrefix+post.PostID.String()), k.Cdc.MustMarshalBinaryBare(&post))
 
 	// Set the last post id
-	store.Set([]byte(types.LastPostIDStoreKey), k.Cdc.MustMarshalBinaryBare(&post.PostID))
+	if post.LastEdited.IsZero() {
```

Comment:
Although this is a good check, I personally don't think it's the best one. 

One case that might be raise is the following: 

- Post with ID 1 is created
- Post with ID 2 is created
- (Somehow) Post with ID 1 is created again

As they are all created, 1 will be again set as the last ID which should not. 

I think there's a better check which is to check whether the ID of the post that is being saved is greater than the last ID saved. If it is, the last ID will be set. Otherwise it will not be set.

Reply:
It make sense, i'll edit it now